lib/ngannotate/processor.rb

```diff
@@ -29,10 +29,14 @@ def force
     # Skip processing in environments where it does not make sense.
     # Override by NG_FORCE=true env variable
     #
-    def skip
+    def skip_by_config
       !force && !::Rails.configuration.ng_annotate.process
     end
 
+    def skip
+      skip_by_config || !@file.start_with?(::Rails.root.to_s)
```

Comment:
I've been doing a bit larger change in config option logic. 

Q: Could you try if this branch handles your deployment case?

https://github.com/kikonen/ngannotate-rails/tree/prepare_only_once

I added also some traces to see what processing is done (see rails logs).